### DISCUSSION TITLE
Need 360° Free Orbit Rotation for 3D Model in All Directions Without Constraints

### DISCUSSION THREAD

**Discussion by RajatDBazaar** [google/model-viewer#4913] [Q&A]
I am developing a 3D model rendering app using Flutter with model-viewer, and everything is working well except for one issue. We need our model to rotate freely in 360° in all directions without restrictions. Currently, the model rotates infinitely on the horizontal axis, but vertically, it is limited to 180°, causing the rotation to lock beyond that point.

What we're aiming for is trackball-style movement, but right now, we only have turntable-style movement. We've been trying various methods to resolve this issue for the past 3 weeks without success, and we're feeling quite exhausted. Any help or suggestions would be greatly appreciated.

Like this website has -- https://app.emb3d.com/3dviewer [View Point - Trackball]

**Comment by elalish** [comment]
You can inspect my [blog post's](https://elalish.blogspot.com/2022/04/3d-interaction.html) JS to see an example, as well as an explanation of why we don't use trackball rotation in MV.

**Comment by RajatDBazaar** [comment]
But what if we want to integrate trackball what things we need to do!!

**Reply by elalish** [reply]
Again:
>You can inspect my [blog post's](https://elalish.blogspot.com/2022/04/3d-interaction.html) JS to see an example

It's not a complicated site.
